lib/jekyll/reader.rb

```diff
@@ -131,12 +132,15 @@ def filter_entries(entries, base_directory = nil)
     # subfolder - The String representing the directory to read.
     #
     # Returns the list of entries to process
-    def get_entries(dir, subfolder)
+    def get_entries(dir, subfolder, limit)
       base = site.in_source_dir(dir, subfolder)
       return [] unless File.exist?(base)
 
       entries = Dir.chdir(base) { filter_entries(Dir["**/*"], base) }
       entries.delete_if { |e| File.directory?(site.in_source_dir(base, e)) }
+
+      entries_limit = [entries.length, limit].min
```

Comment:
The check for positive `limit` in the next line can be hoisted to enclose this line as well and avoid allocating an array when `limit == 0`.

Reply:
to avoid this allocation I'll use trinary operator I see it's used here https://github.com/jekyll/jekyll/blob/master/lib/jekyll/site.rb#L480

Reply:
Fair enough.